sdk/go/common/resource/config/crypt.go

```diff
@@ -112,6 +129,30 @@ func (p panicCrypter) BatchDecrypt(ctx context.Context, ciphertexts []string) ([
 	panic("attempt to bulk decrypt values")
 }
 
+type errorCrypter struct {
+	err string
+}
+
+func NewErrorCrypter(err string) Crypter {
+	return &errorCrypter{err}
+}
+
+func (e errorCrypter) EncryptValue(ctx context.Context, _ string) (string, error) {
+	return "", fmt.Errorf("failed to encrypt: %s", e.err)
+}
+
+func (e errorCrypter) BatchEncrypt(ctx context.Context, _ []string) ([]string, error) {
+	return nil, fmt.Errorf("failed to bulk encrypt: %s", e.err)
+}
+
+func (e errorCrypter) DecryptValue(ctx context.Context, _ string) (string, error) {
+	return "", fmt.Errorf("failed to decrypt: %s", e.err)
+}
+
+func (e errorCrypter) BatchDecrypt(ctx context.Context, _ []string) ([]string, error) {
+	return nil, fmt.Errorf("failed to bulk decrypt: %s", e.err)
```

Comment:
Is `BatchDecrypt` still a required method? I may have missed it -- apologies if so. We should `s/bulk/batch/` if so.

Reply:
Yup - this is now on all crypters, similar to the BatchDecrypt method.